app/main.py

```diff
@@ -31,14 +41,30 @@ def main() -> None:
         Message(toilet_id, MessageType.CLEAN),
     ]
 
-    # run the programs
-    service.run_program(wake_up_program)
-    service.run_program(sleep_program)
+    await sequence_handling(
+        service.run_program(wake_up_program),
+        service.run_program(sleep_program),
+    )
+
+    await parallel_handling(
+        service.unregister_device(hue_light_id),
+        service.unregister_device(speaker_id),
+        service.unregister_device(toilet_id),
+    )
```

Comment:
that's a wrong solution. Basically you don't need to override run_program. In this task you should use a combination of run_parallel / run_sequence, so that the order always will be the same, but with async programming and faster completion